src/components/Events/Events.js

```diff
@@ -6,10 +6,18 @@ import { EventsList, EventFilters, Loading } from '../';
 import { eventsAPI } from '../../api';
 import styles from './Events.sass';
 
+const odataFilterTypes = ['searchText', 'startDate'];
+const nonOdataFiltertypes = ['location', 'range'];
+
 function hasMoreEventsToLoad(currentPage, totalPages) {
   return currentPage + 1 < totalPages;
 }
 
+function buildFilterValues(filters, filterTypes) {
```

Comment:
Feels like all this data munging should be in the api service. You could also include an example as a comment.